### ISSUE TITLE
plane.moveToFront() doesn't work in v7

### ISSUE THREAD

**Issue by ZachSaucier** [martinlaxenaire/curtainsjs#66]
Hey Martin,

It seems that the .moveToFront() function got broken in the transition to v7 (or maybe a more recent update):

v6 demo (works): https://codepen.io/ZachSaucier/pen/8f4d596bf199fea8e55cdecb8d1eca94?editors=0010
v7 demo (broken): https://codepen.io/ZachSaucier/pen/50cec0816cccdbaf4c2c681fbebb6e95?editors=0010

Expected behavior: Images load in Curtains. One second later the dog photo is replaced with the NYC photo. 
What happens: The NYC photo never get placed on top. 

I don't see any note about it in the v7 release notes so I assume it's a bug.

**Comment by martinlaxenaire** [comment]
Hey Zach,

Thanks for reporting this bug. It looks like I broke it all by myself a few days ago - (commit fb3c5001d7c1b4e20dd57e1b0dd01461b7be0475)
Actually, what's funny is that the `moveToFront()` method is now acting like a moveToBack method... :sweat_smile:
Try this, you should get the result you're expecting :

```javascript
setTimeout(() => planes[0].moveToFront(), 1000);
```

Some other quick fixes that should work:
- Set the `transparent` property of the plane you want to be in front to `true` on init.
- Change the plane depthTest value at runtime: `setTimeout(() => planes[1].enableDepthTest(false), 1000);`
- If you want to affect more than one plane, you could also increase their Z relativeTranslation by a really small amount, like `0.01`, it should also work.


Anyway, this bug leads me to two conclusions:
- `moveToFront` should be renamed to something like `renderOnTop`. It was introduced before we were able to actually move the planes along the Z axis but I believe this name no longer makes sense.
- I should probably take a big look at the planes drawing orders (the scene draw stacks) again but this needs to be well thought. To fill the draw stacks array we should take into account the program ID, if there's a render target attached or not, etc. Also we might want to draw the ping pong planes first (as seen in #65).

**Comment by ZachSaucier** [comment]
In my actual project I have multiple planes side by side. Then when one is clicked it scales up to fill the browser. I also play a wave animation (based on your wave effect from your CSS-Tricks article) on the plane so the z component of each pixel is animated up and down. So the ability to make sure one plane renders fully on top of the others is pretty important 🙂 

I tried looping through all of the planes and calling .moveToFront() on all except the current plane but it doesn't seem to fix the issue in my case. Maybe that's because of the z animation? Could that be the case?

```
planes.forEach(p => {
  if(p !== plane) {
    p.moveToFront();
  }
});
```

I don't think that turning off the depthTest will work because I need the clicked plane to be on top regardless of its original order. 
The relative z offset won't work because of the z animation.

**Comment by martinlaxenaire** [comment]
Comparing your plain plane objects in your forEach loop won't work! This, however should do the trick:

```javascript
planes.forEach(p => {
  if(p.uuid !== plane.uuid) {
    p.moveToFront();
  }
});
```

You can also get the old `moveToFront` method result by doing that:

```javascript
function oldMoveToFront(plane) {
    plane.moveToFront();
    plane.enableDepthTest(false); // this is the only thing that was removed from moveToFront method
}

// ...later on

oldMoveToFront(plane);
```

I know those are kind of dirty hacks for now but I don't want to rush into reformating the whole scene rendering order process without thinking a bit about it first. Sorry about that.

Cheers,

**Comment by ZachSaucier** [comment]
I understand. Thanks for the oldMoveToFront function. It works great in my project!

**Comment by martinlaxenaire** [comment]
This should be fixed with v7.3.0 and the new `setRenderOrder()` method: d082f62d3e8e9f98dade1abc1b046c9ef267c278

**Comment by ZachSaucier** [comment]
This [doesn't seem to work](https://codepen.io/ZachSaucier/pen/539e516b95a7426e5343482c796f3dd4?editors=0010) for me... Am I doing something wrong?

**Comment by martinlaxenaire** [comment]
Daaamn, I've just pushed a bug...  What a shame! :sweat:
Should be fixed in v7.3.1 - 37d065baa24a1a30906421117b25e02bedec1564
